app/src/main/java/swati4star/createpdf/util/MergePdf.java

```diff
@@ -58,7 +59,7 @@ protected Void doInBackground(String... pdfpaths) {
             // Create pdf copy object to copy current document to the output merged result file
             mFilename = mFilename + pdfExtension;
             mFinPath = mFinPath + mFilename;
-            PdfCopy copy = new PdfCopy(document, new FileOutputStream(mFinPath));
+            PdfCopy copy = new PdfCopy(document, new FileOutputStream(new File(mFinPath)));
```

Comment:
This is not necessary, the file will get automaticaly created by the FileOutputStream. Only the parent directory not existing is the bug cause you should fix here.